content/posts/0006-build-plan-buildpack.md

```diff
@@ -0,0 +1,94 @@
+---
+title: What is the Build Plan Buildpack and why should I care?
+date: "2021-03-05"
+slug: build-plan-buildpack-exploration
+author: feckhardt
+---
+The [Build Plan buildpack](https://github.com/paketo-community/build-plan)
+gives you very precise control over what buildpacks run during a build and what
+dependencies end up in the final image. Let's talk about how it works and
+why/when you would utilize this kind of functionality.
+
+### What is the Build Plan?
+
+The Build Plan is a part of the Cloud Native Buildpacks (CNB) specification
+that is the format for how buildpacks communicate during the [detect phase](https://paketo.io/docs/buildpacks/#detect-phase) of
+the lifecycle. Buildpacks write out a Build Plan that outlines the
+dependency that the buildpack will either provide or require during its build
+phase. Buildpacks may both require and provide any number of dependencies. For
+detection to pass a buildpacks must have all of the dependencies it requires,
+provided by a buildpack that comes before it in the build order and it needs to
+have all of the dependencies that it provides, required by buildpacks that come
+later in the build order. If either of these conditions is not satisfied
+detection will fail for that buildpack. A more in-depth breakdown of the Build
+Plan can be found in the [CNB
+Spec](https://github.com/buildpacks/spec/blob/main/buildpack.md).
+
+### How does the Build Plan Buildpack work?
+
+The Build Plan Buildpack works by adding a `plan.toml`, which mirrors the
+[requires portion of Build
+Plan](https://github.com/buildpacks/spec/blob/main/buildpack.md#build-plan-toml),
+to the root of your app. The buildpack will read the `plan.toml` and make all
+of the requirements listed in it for you. This means that if you add a
+`plan.toml` to the root of your app and then add the Build Plan buildpack to
+the end of your  buildpack group, you will be able to make any arbitrary
+requirements that you want. Furthermore, if you are using the Build Plan
+Buildpack with Paketo Buildpacks it will allow you to specify when you want the
+dependency to be present whether that is during build-time, at launch-time, or both. You
+can specify this for Paketo Buildpacks because they respect two flags that go
+in the `metadata` field of a requirement. Those two flags are `build` and
+`launch`, `build` makes the dependency available during the build phase of
+subsequent buildpacks in the order, and `launch` makes the dependency available
+in the final running image.
+
+### Why would I want this?
```

Comment:
While the example that is outlined here is maybe more applicable, I think you might wanna start with something even simpler. Say for example, you want to build a container that contains Go. You can't do this with the buildpacks as they exist because `go-dist` doesn't require itself during launch. However, by just adding the `build-plan` buildpack and a `plan.toml`, you can get the buildpack to include its dependency in the launch image. I think after that it starts to make more sense to explain more complicated examples like our integration tests.